src/ThreadedExecutor.h

```diff
@@ -0,0 +1,58 @@
+/* Copyright (C) 2018 European Spallation Source, ERIC. See LICENSE file */
+//===----------------------------------------------------------------------===//
+///
+/// \file
+///
+/// \brief Simple threaded executor to simplify the use of threading in this lib
+/// library.
+///
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <concurrentqueue/concurrentqueue.h>
+#include <functional>
+#include <future>
+#include <memory>
+#include <thread>
+
+class ThreadedExecutor {
+private:
+public:
+  using WorkMessage = std::function<void()>;
+  ThreadedExecutor(bool LowPrioThreadExit = false)
+      : LowPrioExit(LowPrioThreadExit), WorkerThread(ThreadFunction) {}
+  ~ThreadedExecutor() {
+    if (LowPrioExit) {
+      SendLowPrioWork([=]() { RunThread = false; });
+    } else {
+      SendWork([=]() { RunThread = false; });
+    }
+    WorkerThread.join();
+  }
+  void SendWork(WorkMessage Message) { MessageQueue.enqueue(Message); }
```

Comment:
```suggestion
  void SendWork(WorkMessage const &Message) { MessageQueue.enqueue(Message); }
```

Reply:
In this case, I think the use of `std::move` is the better solution.